verifier/verifier.go

```diff
@@ -509,6 +543,59 @@ func (v *verifier) processSignature(ctx context.Context, sigBlob []byte, envelop
 	return nil
 }
 
+func (v *verifier) verifyRevocation(ctx context.Context, outcome *notation.VerificationOutcome) *notation.ValidationResult {
+	logger := log.GetLogger(ctx)
+
+	if v.revocationCodeSigningValidator == nil && v.revocationClient == nil {
+		return &notation.ValidationResult{
+			Type:   trustpolicy.TypeRevocation,
+			Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+			Error:  fmt.Errorf("unable to check revocation status, code signing revocation validator cannot be nil"),
+		}
+	}
+
+	var authenticSigningTime time.Time
+	if outcome.EnvelopeContent.SignerInfo.SignedAttributes.SigningScheme == signature.SigningSchemeX509SigningAuthority {
+		authenticSigningTime, _ = outcome.EnvelopeContent.SignerInfo.AuthenticSigningTime()
+	}
```

Comment:
This is for my understanding I think. Why are we considering the authentic signing time only in the case of signing authority but not in notary.x509 case?

Reply:
This PR is not changing the original logic at this part. But I think I could explain more. This is according to the Notary Project spec:  https://github.com/notaryproject/specifications/blob/main/specs/signature-specification.md#standard-attributes
It says:
```
Authentic Signing Time (critical):  ... This claim is REQUIRED and only valid when signing scheme is notary.x509.signingAuthority. ....
```

Reply:
I understand that AuthenticSigningTime only applies to `notary.x509.signingAuthority`. But, the equivalent of it in the case of `notary.x509` is the RFC#3161 TSA signature. My question is why are we not taking the TSA signature into account (for the case of `notary.x509` signatures) for revocations?

Reply:
@rgnote Regarding revocations, we've decided to exclude the TSA signature consideration for now. The main reason is that revocation timing is not deterministic, which may not accurately reflect the actual date of invalidation/compromising.

Reply:
`AuthenticSigningTime` and `RFC#3161 TSA signature` are not equivalent though. `RFC#3161 TSA signature` does not provide information about the `signing time`, it's just the time point when the TSA sees the signature. However, `AuthenticSigningTime` is the exact `signing time`.
Another difference is that `AuthenticSigningTime` is a signed attribute. However, `TSA signature` is an unsigned attribute, if we want to use it, we MUST verify the signature before consuming it. Note: Since the `Authentic Timestamp` step can be set to `logged` and `timestamp verification` can be disabled by trust policy, we cannot directly consume it during `Revocation` step.
Therefore, in this version, we tend to not use it during `Revocation`, in this way, whenever the cert status is `revoked`, we just say the cert chain is revoked (this is more strict, so there isn't security concern).

Reply:
> AuthenticSigningTime and RFC#3161 TSA signature are not equivalent though. RFC#3161 TSA signature does not provide information about the signing time, it's just the time point when the TSA sees the signature. However, AuthenticSigningTime is the exact signing time.

RFC#3161 addresses this by rejecting timestamps if they are way too far from the actual signing time. Also, AuthenticSigningTime doesn't necessarily be the exact signing time. It can be set to any timestamp by a Signing Authority (due to a bug for example). For all practical purposes, TSA signature and AuthenticSigingTime are equivalent. The only difference is "who" is attesting those timestamps.


However, thanks for the explanation though. I acknowledge the complexity of using TSA signatures in conjunction with  revocation effective time. However, I don't see this being documented anywhere. I'm worried about users revoking their certs with effective time and assuming that Notation would correctly pass (or fail) signature verification for notary.x509 signatures. If notation is not going to support it, we should atleast call this out in our documentation and also track this as an issue for future consideration. Can we please track this as an issue that Notation does not consider revocation effective time for notary.x509 signatures due to X,Y and Z reasons.  Thanks.

Reply:
@rgnote Thanks. I created a new issue https://github.com/notaryproject/notation-go/issues/436 for tracking.
We already had a document issue, and I added a comment to call out the revocation scenarios. https://github.com/notaryproject/notaryproject.dev/issues/404. Hope it works for you.